core/crypto/src/signer.rs

```diff
@@ -2,13 +2,14 @@ use crate::key_conversion::convert_secret_key;
 use crate::key_file::KeyFile;
 use crate::{KeyType, PublicKey, SecretKey, Signature};
 use near_account_id::AccountId;
+use serde::{Deserialize, Serialize};
 use std::fmt::{self, Debug};
 use std::io;
 use std::path::Path;
 use std::sync::Arc;
 
 /// Enum for Signer, that can sign with some subset of supported curves.
-#[derive(Debug, PartialEq)]
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
```

Comment:
Why do you need serialize and deserialize?

Reply:
It is needed in the next PR. So I will remove it here and add to the next PR instead.